spec/lib/logging/third_party_transaction_spec.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Logging::ThirdPartyTransaction do
+  # to use shared examples
+  # 1. define variable test_object of desired type using a `let` block
+  # 2. test_object must define an instance method `method_to_wrap`
+  # 3. test_object should extend Logging::ThirdPartyTransaction::MethodWrapper
+  # 4. test_object definition should end with a call to .new, ensuring an object
+  #    is returned to your variable, rather than the class definition block,
+  #    as our linter will reject this.
+  shared_examples_for 'a third party transaction logger' do
```

Comment:
(Non blocking) I forgot about shared examples this is handy. For only a handful of cases this feels like a little overkill  IMO since it requires the reader to understand the pattern you are using here but presumably if you want to test how this module works with other objects there will be more cases here and DRYing it all up makes sense

Reply:
Yeah, shared examples are arguably an anti-patern, depending on who you ask, given the extra layer of abstraction and Ruby's stated mission of being readable.  But, in this case, it feels right?  ¯\_(ツ)_/¯